### ISSUE TITLE
TO /servers/{hostname}/update_status parentRevalPending is not CDN-specific

### ISSUE THREAD

**Issue by rawlinp** [apache/trafficcontrol#5505]
## I'm submitting a ...
-  bug report

## Traffic Control components affected ...
-  Traffic Ops

## Current behavior:
If you have two CDNs (A and B) and a `MID` in CDN A (cachegroup = parent-cg) with `revalPending = true`, all child caches of cachegroup parent-cg in both CDN A and B have `parentRevalPending = true`.

## Expected behavior:
In the above example, only child caches of cachegroup parent-cg _that are in CDN A_ would have `parentRevalPending = true`. 

## Minimal reproduction of the problem with instructions:
Using the above example, ensure that CDN B has no pending revalidations then submit an invalidation for a delivery service in CDN A. Observe that child caches in CDN B now have `parentRevalPending = true`.

**Comment by ocket8888** [comment]
I just tried to reproduce this in CDN-in-a-Box. I created a new CDN - B - and cloned an Edge Server Profile into it. Then I created a new server using that Profile (obviously, then, the server was also in CDN B) and put it into the same Cache Group as the default Edge Server (`edge`), and finally created a content invalidation job on `demo1`. Afterwards, `edge`, `mid-01`, and `mid-02` all had `reval_pending` values of `true`. The new Edge Server in CDN B did not.

Is this because I was using a Topology? Does the problem only manifest with traditional server assignments? Or did I miss a step somewhere?
